### DISCUSSION TITLE
Filtering Expressions

### DISCUSSION THREAD

**Discussion by indy1725** [Athou/commafeed#1123] [General]
Is it just me, or are filtering expressions not working 100% of the time?

The example I'm working on is with the BGR feed: https://bgr.com/feed/
I'm trying to filter out explicitly any post with \<category>\<![CDATA[BGR Deals]]>\</category>
and my filtering expression is `(not categories.contains('bgr deals'))`

But I still get the BGR Deals posts showing up in my feed. What am I doing wrong here?

I am linking multiple `(not ... )` statements with 'or' between them, but if any of them returns 'false' then the whole thing should fail, right?

To exclude multiple categories, should I use `(not categories.contains(['bgr deals', 'whatever', whichever']))` instead?

**Comment by Athou** [comment]
> I am linking multiple `(not ... )` statements with 'or' between them, but if any of them returns 'false' then the whole thing should fail, right?

If you want the expression to return false if any condition is false, you need to use `and` instead of `or`: you want CommaFeed to show the entry if category A is not present AND category B is not present.

**Reply by indy1725** [reply]
Okay, yeah I can see that now. I think what was throwing me was that I was checking for False instead of checking for True.
